### PR TITLE
Feature/pi 686 add an as

### PR DESCRIPTION
**Pull request by megan-bower4**

### CODE REVIEW

File: src/layers/sds/epr/creators.py [hunk 1]
```diff
@@ -94,14 +94,12 @@ def create_as_device(
...
     message_sets_id: str,
     additional_interactions_id: str,
-    as_tags: list[dict],
 ) -> Device:
     as_device = product.create_device(
         name=EprNameTemplate.AS_DEVICE.format(party_key=party_key, asid=asid)
     )
     as_device.add_key(key_type=DeviceKeyType.ACCREDITED_SYSTEM_ID, key_value=asid)
     as_device.add_questionnaire_response(as_device_data)
-    as_device.add_tags(tags=as_tags)
```
Comment (jaklinger) [inline src/layers/sds/epr/creators.py:104]: don't think we want to remove tags do we?

File: src/layers/sds/epr/utils.py [hunk 2]
```diff
@@ -32,7 +32,15 @@ def is_as_device(device: Device) -> bool:
...
-    return {qr.data[SdsFieldName.INTERACTION_ID] for qr in questionnaire_responses}
+
+    # Handle the case where questionnaire_responses is empty
+    if not questionnaire_responses:
+        return set()
+
+    return {
+        qr.data[SdsFieldName.INTERACTION_ID]
+        for qr in next(iter(questionnaire_responses), [])
+    }
```
Comment (jaklinger) [inline src/layers/sds/epr/utils.py:46]: fwiw an implicit way (and also preventing multiple `return`) would be

```python
    try:
        (questionnaire_responses,) = (
            message_sets_or_additional_interactions.questionnaire_responses.values()
        )
    except ValueError:
        interaction_ids = set()
    else:
        interaction_ids = {qr.data[SdsFieldName.INTERACTION_ID] for qr in questionnaire_responses}
    return interaction_ids
```

File: src/layers/sds/epr/creators.py (hunk 1 above)
Comment (megan-bower4) [reply src/layers/sds/epr/creators.py:104]: Ah I was following the pattern for mhs, as they were removed there, and I thought they'd be captured in the tags ticket. But do AS Devices still need the tags of the sds fhir query params? And mhs have no tags then?

Comment (jaklinger) [reply src/layers/sds/epr/creators.py:104]: yeah AS tags are still needed for SDS FHIR API - but there are fewer of them, and MHS devices don't have tags any more
=

**Review by jaklinger — APPROVED** [review]

**Review by jaklinger — APPROVED** [review]

**Pull request merged**